lib/model/rofolder.go

```diff
@@ -16,113 +16,135 @@ import (
 
 type roFolder struct {
 	stateTracker
+	scan
```

Comment:
You're extracting the scan related functionality from roFolder into a type of it's own. The same functionality is essentially duplicated in rwFolder. If we're factoring it out, it would be nice to have it reused between the two as well.